### ISSUE TITLE
Error when starting vite dev server

### ISSUE THREAD

**Issue by YKDZ** [vikejs/vike#1986]
### Description

When upgrade vike from 0.4.199 to 0.4.200, I have met this issue when running `pnpm dev`:
```
failed to load config from D:\Web App Project\fungi\vite.config.ts
error when starting dev server:
ReferenceError: exports is not defined in ES module scope
    at file:///D:/Web%20App%20Project/fungi/node_modules/.pnpm/@brillout+vite-plugin-server-entry@0.5.2/node_modules/@brillout/vite-plugin-server-entry/dist/runtime/crawlServerEntry.js:2:23
    at ModuleJobSync.runSync (node:internal/modules/esm/module_job:366:35)
    at ModuleLoader.importSyncForRequire (node:internal/modules/esm/loader:325:47)
    at loadESMFromCJS (node:internal/modules/cjs/loader:1381:24)
    at Module._compile (node:internal/modules/cjs/loader:1503:5)
    at Object..js (node:internal/modules/cjs/loader:1689:10)
    at Module.load (node:internal/modules/cjs/loader:1318:32)
    at Function._load (node:internal/modules/cjs/loader:1128:12)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
 ELIFECYCLE  Command failed with exit code 1.
```
Here is my `package.json`:
```
{
  "name": "fungi",
  "version": "0.0.1",
  "description": "Home site & Yggdrasil API implements for Forest Realm",
  "type": "module",
  "scripts": {
    "dev": "vite dev",
    "build": "vite build",
    "preview": "cross-env NODE_ENV=production node ./dist/server/index.mjs",
    "lint": "eslint . --fix",
    "prisma:studio": "prisma studio",
    "prisma:generate": "prisma generate",
    "prisma:dev": "prisma migrate dev",
    "prisma:deploy": "prisma migrate deploy",
    "prisma:seed": "tsx prisma/seed.ts",
    "docker:build": "docker build . -f ./docker/Dockerfile -t ykdz/fungi:latest",
    "docker:push": "docker push ykdz/fungi:latest"
  },
  "keywords": [],
  "author": "YKDZ",
  "devDependencies": {
    "@eslint/js": "^9.15.0",
    "@iconify-json/carbon": "^1.2.4",
    "@iconify-json/lucide": "^1.2.15",
    "@iconify-json/mdi": "^1.2.1",
    "@iconify-json/vscode-icons": "^1.2.2",
    "@types/aws-lambda": "^8.10.145",
    "@types/bcryptjs": "^2.4.6",
    "@types/cookie": "^0.6.0",
    "@types/jsonwebtoken": "^9.0.7",
    "@types/markdown-it": "^14.1.2",
    "@types/node": "^22.9.3",
    "@types/nodemailer": "^6.4.17",
    "@types/qrcode": "^1.5.5",
    "@types/speakeasy": "^2.0.10",
    "@unocss/autocomplete": "^0.64.1",
    "@unocss/eslint-config": "^0.64.1",
    "@unocss/postcss": "^0.64.1",
    "@unocss/preset-attributify": "^0.64.1",
    "@unocss/preset-icons": "^0.64.1",
    "autoprefixer": "^10.4.20",
    "eslint": "^9.15.0",
    "eslint-plugin-vue": "^9.31.0",
    "prisma": "^5.22.0",
    "ts-to-zod": "^3.13.0",
    "typescript": "^5.7.2",
    "typescript-eslint": "^8.15.0",
    "unocss": "^0.64.1",
    "unplugin-vue-components": "^0.27.4",
    "vite": "^5.4.11",
    "vite-plugin-cjs-interop": "^2.1.5",
    "vite-plugin-pwa": "^0.21.0",
    "vue-eslint-parser": "^9.4.3"
  },
  "dependencies": {
    "@azure/msal-browser": "^3.27.0",
    "@azure/msal-node": "^2.16.2",
    "@hono/node-server": "^1.13.7",
    "@hono/node-ws": "^1.0.4",
    "@prisma/client": "^5.22.0",
    "@trpc/client": "11.0.0-rc.502",
    "@trpc/server": "11.0.0-rc.502",
    "@types/ua-parser-js": "^0.7.39",
    "@universal-middleware/core": "^0.2.13",
    "@universal-middleware/hono": "^0.2.12",
    "@unocss/reset": "^0.64.1",
    "@vitejs/plugin-vue": "^5.2.0",
    "@vue/compiler-sfc": "^3.5.13",
    "@vue/server-renderer": "^3.5.13",
    "@vueuse/core": "^11.3.0",
    "@vueuse/integrations": "^11.3.0",
    "base64url": "^3.0.1",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.1.1",
    "cron": "^3.2.1",
    "cross-env": "^7.0.3",
    "cross-fetch": "^4.0.0",
    "dotenv": "^16.4.5",
    "hono": "^4.6.11",
    "libsodium": "^0.7.15",
    "markdown-it": "^14.1.0",
    "nodemailer": "^6.9.16",
    "otpauth": "^9.3.5",
    "pinia": "^2.2.6",
    "pinia-plugin-persistedstate": "^4.1.3",
    "qrcode": "^1.5.4",
    "radix-vue": "^1.9.10",
    "redis": "^4.7.0",
    "skinview3d": "^3.1.0",
    "tsx": "^4.19.2",
    "ua-parser-js": "^1.0.39",
    "universal-cookie": "^7.2.2",
    "unplugin-vue-markdown": "^0.26.2",
    "vike": "^0.4.199",
    "vike-node": "^0.1.16",
    "vike-vue": "^0.8.5",
    "vue": "^3.5.13",
    "zod": "^3.23.8"
  },
  "imports": {
    "@/*": "./src/*.ts"
  },
  "packageManager": "pnpm@9.14.2"
}
```

**Comment by brillout** [comment]
Try with 0.4.204

**Comment by YKDZ** [comment]
```
failed to load config from D:\Web App Project\fungi\vite.config.ts
error when starting dev server:
ReferenceError: exports is not defined in ES module scope
    at file:///D:/Web%20App%20Project/fungi/node_modules/.pnpm/@brillout+vite-plugin-server-entry@0.5.2/node_modules/@brillout/vite-plugin-server-entry/dist/runtime/crawlServerEntry.js:2:23
    at ModuleJobSync.runSync (node:internal/modules/esm/module_job:366:35)
    at ModuleLoader.importSyncForRequire (node:internal/modules/esm/loader:325:47)
    at loadESMFromCJS (node:internal/modules/cjs/loader:1381:24)
    at Module._compile (node:internal/modules/cjs/loader:1503:5)
    at Object..js (node:internal/modules/cjs/loader:1689:10)
    at Module.load (node:internal/modules/cjs/loader:1318:32)
    at Function._load (node:internal/modules/cjs/loader:1128:12)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
 ELIFECYCLE  Command failed with exit code 1.
```
Here is the error stack after upgrading to 0.4.204

**Comment by brillout** [comment]
Let me try to reproduce.

Related: https://github.com/brillout/vite-plugin-server-entry/issues/14.

**Comment by brillout** [comment]
[Fix](https://github.com/brillout/vite-plugin-server-entry/commit/36d6cbba88f155b91d9dd57c16d7fa8b69a72241) released in `@brillout/vite-plugin-server-entry@0.5.3`. Update your lock file.
